apps/emqx_s3/src/emqx_s3_client.erl

```diff
@@ -273,10 +278,10 @@ request_fun(HttpPool, PoolType, MaxRetries) ->
         end)
     end.
 
-ehttpc_request(HttpPool, Method, Request, Timeout, MaxRetries) ->
-    try timer:tc(fun() -> ehttpc:request(HttpPool, Method, Request, Timeout, MaxRetries) end) of
+ehttpc_request(Worker, Method, Request, Timeout, MaxRetries) ->
```

Comment:
Q: given that we inject a http request func from config, make sense to delete all the http client dependencies from our erlcloud fork ?

Reply:
I think we still use the default client (hackney) in `emqx_bridge_dynamo` and in tests. Besides, `ehttpc_pool`-based approach won't currently work smoothly if we'd want support for vhost-only bucket access, supposedly the easiest way would be to switch back to hackney (though not the only way).

Reply:
ok, at least lhttpc is not needed https://github.com/emqx/erlcloud/blob/2e95901cfd434e05ac6871d5a3134dd757b2cefd/rebar.config#L21 ?